### ISSUE TITLE
[Feature] Support for sortType and sortOrder in outputColumns

### ISSUE THREAD

**Issue by Udayraj123** [Udayraj123/OMRChecker#209]
**Is your feature request related to a problem? Please describe.**
Currently we support `outputColumns` to be custom defined and we sort in an ascending alphanumerical order by default. 
The users may want to dynamically sort the output columns in a descending order or just alphabetical instead of alphanumerical order.

**Describe the solution you'd like**
We sort the output columns [here](https://github.com/Udayraj123/OMRChecker/blob/bf19841b548d4181ff304cad75826299128f38cc/src/template.py#L123) using the function custom_sort_output_columns.
- We can modify this to accept user input for `outputColumns.sortType`(ALPHABETICAL, ALPHANUMERIC, CUSTOM) and `outputColumns.sortOrder`(ASC/DESC) in the template.json
- In case of CUSTOM we expect the user to enter all available columns in `outputColumns.columns`
- Make sure to do corresponding changes in the template_schema.py file and add its validations

**Describe alternatives you've considered**
As of now user has to define the custom order in case of descending order. Which may not be possible in case of custom sets(v2)

**Additional context**
N/A

Note: please share your queries and approaches on discord for quicker discussions:
- Link to the [Discord invite](https://discord.gg/qFv2Vqf)
- Link to the [Hacktoberfest 2024 channel](https://discord.com/channels/590134763784896514/1284801897714155613/1284802010549321770)

**Comment by harshsennnn** [comment]
can u assign me this issue

**Comment by Anushlinux** [comment]
Hello, I have a solution in my mind for outputColumns.sortOrder  can you assign this issue to me?

**Comment by Udayraj123** [comment]
Hey @harshsennnn and @Anushlinux, 
I will have to assign on a first come first serve basis - but only once one of you shares and discusses an approach that looks feasible.

If you don't get assigned don't worry there are other good first issues to look at, especially submitting your first sample via https://github.com/Udayraj123/OMRChecker/issues/40

Also you can join the discord server to discuss any queries with the community.

**Comment by Prasadayus** [comment]
Proposed Solution:
Enhanced Sorting Options: Modify the sorting functionality of custom_sort_output_columns to accept user-defined sorting preferences via the template.json file. 
Specifically:

- Add an option for outputColumns.sortType that supports the following values:

2.ALPHABETICAL: Sort the columns alphabetically.
3.ALPHANUMERIC: Default alphanumerical sorting.
4.CUSTOM: User defines a custom order.

- Add an option for outputColumns.sortOrder that supports:
1.ASC: Ascending order (default behavior).
2.DESC: Descending order.

- For the CUSTOM option, users should specify the exact column order in outputColumns.columns.

 Schema Changes: Update template_schema.py to reflect these new fields and validate the user inputs:

1.Ensure that sortType accepts only the allowed values (ALPHABETICAL, ALPHANUMERIC, CUSTOM).
2.Ensure that sortOrder accepts only ASC or DESC.
3.For the CUSTOM type, validate that all available columns are listed in the custom order.

@Udayraj123 can I work on this ?

**Comment by Udayraj123** [comment]
@YashBaviskar1 that looks good. You can work on this feature. Meanwhile @Prasadayus I see you've commented on one more issue, you can work on that instead.

**Comment by JoyChopra1298** [comment]
@YashBaviskar1 did you get a chance to implement this? 

@Udayraj123 Or is it still open to be picked?

**Comment by Udayraj123** [comment]
@JoyChopra1298 Yes the issue is still open with no active PRs. Shall I assign it to you? please share a basic approach before you start working

**Comment by JoyChopra1298** [comment]
Yes, I can pick up this issue. You can assign it to me. My approach is similar to what is previously stated.

I plan to add if statements to check for the 3 sort types and accordingly set the output columns. I will use sort order to determine the value of reversed parameter in sorted function. This should support the usecase as requested.

**Comment by Udayraj123** [comment]
Should close automatically once dev branch is merged into master
